src/frontend/screens/Onboarding/AddToProjectScreen/index.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from "react";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import { View, StyleSheet } from "react-native";
+import { FormattedMessage, defineMessages } from "react-intl";
+
+import { MEDIUM_BLUE, WHITE } from "../../../lib/styles";
+import HeaderTitle from "../../../sharedComponents/HeaderTitle";
+import { BackIcon } from "../../../sharedComponents/icons";
+import IconButton from "../../../sharedComponents/IconButton";
+import { WithWifiBar } from "../WithWifiBar";
+
+import { ScanQrCodeStep } from "./ScanQrCodeStep";
+import { DeviceFoundStep } from "./DeviceFoundStep";
+import { SuccessStep } from "./SuccessStep";
+
+type Step = "scan" | "found" | "success";
+
+const m = defineMessages({
+  title: {
+    id: "screens.Onboarding.AddToProjectScreen.title",
+    defaultMessage: "Add to {projectName}",
+  },
+});
+
+export const AddToProjectScreen: NavigationStackScreenComponent = () => {
+  const [step, setStep] = React.useState<Step>("scan");
+  const [foundDeviceId, setFoundDeviceId] = React.useState<string>();
+
+  // TODO: Get the actual project name here
+  const projectName = "BarFoo";
+
+  const getRenderedStep = () => {
+    switch (step) {
+      case "scan":
+        return (
+          <ScanQrCodeStep
+            goNext={data => {
+              setFoundDeviceId(data);
+              setStep("found");
+            }}
+          />
+        );
+      case "found":
+        return (
+          !!foundDeviceId && (
+            <DeviceFoundStep
+              deviceId={foundDeviceId}
+              goNext={() => setStep("success")}
+              goBack={() => {
+                setFoundDeviceId(undefined);
+                setStep("scan");
+              }}
+            />
+          )
+        );
+      case "success":
+        return (
+          !!foundDeviceId && (
+            <SuccessStep
+              goNext={() => {
+                setFoundDeviceId(undefined);
+                setStep("scan");
+              }}
+              deviceId={foundDeviceId}
+              projectName={projectName}
+            />
+          )
+        );
+    }
+  };
+
+  return (
+    <WithWifiBar>
+      <View style={styles.container}>{getRenderedStep()}</View>
+    </WithWifiBar>
+  );
+};
+
+AddToProjectScreen.navigationOptions = () => ({
+  animationEnabled: false,
+  headerTitle: () => (
+    <HeaderTitle style={{ color: WHITE }}>
+      {/* TODO: Get the actual project name here */}
+      <FormattedMessage {...m.title} values={{ projectName: "BarFoo" }} />
+    </HeaderTitle>
```

Comment:
The name is available through 
* `const [config] = useContext(ConfigContext);` 
* `const name = config.metadata.name`

Reply:
Oh I just realized that you are trying to access it OUTSIDE of the component. I havent quite figured out how to do this in v4 of react nav

Reply:
Think you can still use the context in this case since the navigation lives as a child of the config provider, so what you proposed should work. One small detail I'm a little unsure about is it seems that `config.metadata.name` seems to be `string | undefined`, so should we worry about when it's `undefined`, or is it safe to assume that it'll always exist and we just need to make TS happy with nevery-reaching fallbacks?

Reply:
> Think you can still use the context in this case since the navigation lives as a child of the config provider, so what you proposed should work.

nvm just tried to use the context in the header title and it didn't work, so i guess that's an unknown

Reply:
guess the only reasonable thing i see is setting the navigation params from the main component and using that in the header title, which seems meh 🤷‍♂️

Reply:
Yeah, I wouldn't worry too much about it rn, i think there is a slightly different pattern in v6 that makes it easier

Reply:
I was dealing with the undefined by just conditionally changing it to be generic: Eg if(undefined): "Leave Project" vs if("name"): "Leave Project name"

Reply:
Accessing context in the header is a pain in RN@4. You need to use navigation params see https://reactnavigation.org/docs/4.x/header-buttons

Hopefully this is easier in v6 and we can go directly to that.